### PR TITLE
Modernize tutorial final code

### PR DESCRIPTION
**Pull request by rjrjr**
It's time to get the Tutorial caught up with undeprecated API. This PR modifies only the final tutorial module. Once we're happy with the code, I'll back port the changes and update the prose in follow ups.

- Use `AndroidScreen` and drop `ViewRegistry`
- Better, more consistent use of `BackStackScreen`
- Use `View.setBackHandler`
- Use `TextController`
- Delete a lot of `// Exactly what the function name and the code say` comments
- More consistent naming, code style for actions and event handlers in `Screen` renderings
  - Event handler fields are named `onVerbPhrase`, like `onBackPressed`
  - Action names are verb phrases describing the action that is being taken, not the event that is being handled
  - Output names are generally in terms of the semantic event being reported to the parent, rather than describing what the parent will do

Thus:

```kotlin
return TodoListScreen(
  onRowPressed = { context.actionSink.send(reportSelection(it)) }
```

```kotlin
  private fun reportSelection(index: Int) = action {
    // Tell our parent that a todo item was selected.
    setOutput(TodoSelected(index))
  }
```

I did not introduce `RenderContext.eventHandler {}`, that seems like it would just be confusing to a newcomer.

Also not introducing big new blocks of material, in particular not introducing `Overlay`. I do think we should do that, but for this release I just want to focus on getting the deprecated code deleted.

### CODE REVIEW

File: workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/container/BackStackScreen.kt [hunk 1]
```diff
@@ -115,6 +115,13 @@ public operator fun <T : Screen> BackStackScreen<T>.plus(
   return other?.let { BackStackScreen(frames + it.frames) } ?: this
 }
 
+@WorkflowUiExperimentalApi
+public operator fun <T : Screen> BackStackScreen<T>.plus(
+  other: BackStackScreen<T>?
+): BackStackScreen<T> {
+  return other?.let { BackStackScreen(frames + it.frames) } ?: this
+}
```
Comment (steve-the-edwards) [inline workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/container/BackStackScreen.kt:123]: Could separate this out to a different PR that's not tutorial related. but also 🤷🏻

**Review by steve-the-edwards — APPROVED** [review]
LGTM. Still don't feel like the way we patternize `action` creation is ideal - but I'm not sure what I offered is better.

Also, if you have a backlog of what else to add to the tutorial can I request:
1. A SessionWorkflow
2. Other testing types - like `headlessIntegrationTest` or `launchForTestingFrom...`?

File: workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/container/BackStackScreen.kt (hunk 1 above)
Comment (rjrjr) [reply workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/container/BackStackScreen.kt:123]: Yeah, this seems like it leaked in from another PR (that I thought I had already merged).

**Comment by rjrjr** [comment]
> Also, if you have a backlog of what else to add to the tutorial can I request:
> 
> 1. A SessionWorkflow
> 2. Other testing types - like `headlessIntegrationTest` or `launchForTestingFrom...`?

We need proper code labs. And more people. :/

**Comment by rjrjr** [comment]
Merging to the staging `ray/tutorial-update` branch. Red shards are current noise plus a compile error in tutorial 2, which hasn't been updated yet.
